packages/collectxyz/src/nft.rs

```diff
@@ -103,9 +146,53 @@ impl XyzExtension {
     pub fn has_arrived(&self, current_time: Timestamp) -> bool {
         self.arrival <= current_time
     }
+
+    pub fn as_traits(&self) -> Vec<Cw721Trait> {
```

Comment:
We don't want to include previous coordinates and arrival in here? I suppose it's a necessary just a thought

Reply:
I don't feel strongly on this, but kind of like the minimalism of only including current coordinates.